### ISSUE TITLE
cron: borg > /dev/null is not quiet

### ISSUE THREAD

**Issue by breyer** [borgbackup/borg#425]
Running
``/usr/local/bin/borg create -foo -bar > /dev/null``
with cron is not quiet.

Output is:
``reading files cache``
``processing files``

Running
``/usr/local/bin/borg create -foo -bar > /dev/null 2>&1``
works, but suppresses errors.

**Comment by ThomasWaldmann** [comment]
because we support writing "borg extract" data to stdout, we do no user / log output via stdout, but via stderr.

Next release will likely have logging configuration, so you can configure it to also output timestamp and level (INFO, WARNING, ERROR) and raise the minimum log level to WARNING, if you like.

**Comment by ThomasWaldmann** [comment]
hmm, maybe we should also have a commandline option that sets the minimum log level. This is currently hardcoded to INFO, but it seems you'ld prefer WARNING.

**Comment by anarcat** [comment]
@ThomasWaldmann this is what i attempted in the original logging refactoring run. i set the default log level to WARNING and made multiple `-v` flags bump it up to INFO then DEBUG.

in my opinion, borg should be silent by default. but we could also have a `--quiet` flag to bump the level up to WARNING and silence the info messages.

**Comment by anarcat** [comment]
this reminds me... it looks like we're lying here: http://borgbackup.readthedocs.org/en/stable/usage.html#quiet-by-default - another SNAFU of the logging-refactor merge.

**Comment by ThomasWaldmann** [comment]
we already discussed that. -v is because one wants more informative output, not because one wants to see developer/debug messages. so verbosity is a sematically different thing from the log level.

**Comment by anarcat** [comment]
well, i remember we disagreed on that, for sure. what i am saying is that borg is *not* currently "quiet by default" as documented.

**Comment by ThomasWaldmann** [comment]
that might be. it even engages progress indication automatically. :-P

**Comment by anarcat** [comment]
well, progress will shutup on cronjobs for sure. that was designed that way. "silent by default" for me is important exactly for the reasons stated here (be able to silence the script without shutting down warnings), and i had a good working solution for it (using -v to control the log level).

failing other solutions, i suggest we implement that one, even if it's not exactly what you had in mind.

**Comment by ThomasWaldmann** [comment]
I guess it would be useful to control the log level by commandline argument (for cases not using a logging.conf file), we can use e.g. "--log-level warning" for that (and stay away from -v).

**Comment by anarcat** [comment]
that's a little verbose. and i would then have no idea what `-v` would be for: right now it doesn't really do anything...

**Comment by ThomasWaldmann** [comment]
It does, see borg create -v. I am repeating myself...

**Comment by anarcat** [comment]
borg create -v floods the terminal with all files in the archive, from what i remember. it's pretty much all that -v does right now. in my mind, that (listing all files) should be a topical flag (see #302) but i am also repeating myself as well... i guess i don't understand what you want `-v` to be doing...

> -v is because one wants more informative output, not because one wants to see developer/debug messages

If the default logging level was `WARNING` and `-v` would enable `INFO`, that would do exactly what you are saying here. We could have a `--debug` flag to enable `DEBUG` as well. if you dislike the idea of multiple `-v` flags enabling debug...

I see no use case for a `--log-level`: why would a user want to disable `WARNING` or `ERROR` messages exactly?

i find that having two sets of semantically similar commandline flags to control output is confusing. what would `-v --log-level WARNING` mean?

in summary, what i would propose is:

* default: log level WARNING, ERROR, and up
* `-v`: log level INFO and up
* `-vv` or `-d / --debug`: log level DEBUG and up

**Comment by anarcat** [comment]
i'd of course be happy to implement the above. i don't think i can fix this issue otherwise.

**Comment by ThomasWaldmann** [comment]
I've said (here or in other tickets where this was discussed) all what I think is needed to see why log levels are not to be confused with verbosity or selection of informative topics one wants to output.
Still you insist modifying log level by -v. I can't follow, sorry.

Also, as this is obviously leading nowhere, I am closing this issue.

**Comment by anarcat** [comment]
i think selective information topics like `--progress`, `--stats` and similar are fine. i am fine with the outcome of #302.

i don't think `--verbose` is a topical flag. it is a generic flag that is used in hundreds of other commandline tools. i don't understand what it means if it doesn't apply to log levels: it is too generic to be meaningful, and it is the problem we are seeing right now, in this very issue, and others.

**Comment by anarcat** [comment]
@breyer your original issue will be resolved by #426, eventually.

**Comment by breyer** [comment]
Would be great. Thanks.
